platform/posix/transport/src/openssl_posix.c

```diff
@@ -0,0 +1,563 @@
+/* Standard includes. */
+#include <assert.h>
+#include <stdlib.h>
+#include <string.h>
+
+/* POSIX socket includes. */
+#include <limits.h>
+#include <errno.h>
+#include <netdb.h>
+#include <time.h>
+#include <poll.h>
+#include <unistd.h>
+#include <arpa/inet.h>
+
+#include <sys/socket.h>
+#include <sys/types.h>
+
+/* OpenSSL include. */
+#include <openssl/ssl.h>
+
+/* Transport interface include. */
+#include "transport_interface.h"
+
+#include "openssl_posix.h"
+
+/*-----------------------------------------------------------*/
+
+/**
+ * @brief Timeout for transport send.
+ *
+ * @note Setting to a negative value implies an infinite timeout.
+ */
+static int tlsSendTimeout = -1;
```

Comment:
What if I want multiple instances of this transport interface with differing timeout values, e.g. one WiFi and one cellular?

Reply:
As discussed with @abhidixi11, we are handling this problem by allowing the user to define the timeout by passing a parameter to `TCP_Connect`. Essentially, that sets a timeout for the socket using `setsockopt`.